src/dpp/discordvoiceclient.cpp

```diff
@@ -1165,20 +1167,29 @@ uint32_t discord_voice_client::get_tracks_remaining() {
 
 discord_voice_client& discord_voice_client::skip_to_next_marker() {
 	std::lock_guard<std::mutex> lock(this->stream_mutex);
-	/* Keep popping the first entry off the outbuf until the first entry is a track marker */
-	while (!outbuf.empty() && outbuf[0].packet.size() != sizeof(uint16_t) && (*((uint16_t*)(outbuf[0].packet.data()))) != AUDIO_TRACK_MARKER) {
-		outbuf.erase(outbuf.begin());
-	}
-	if (outbuf.size()) {
-		/* Remove the actual track marker out of the buffer */
-		outbuf.erase(outbuf.begin());
+	if (!outbuf.empty()) {
+		/* Find the first marker to skip to */
+		auto i = std::find_if(outbuf.begin(), outbuf.end(), [](const voice_out_packet &v){
+				return v.packet.size() == sizeof(uint16_t) && (*((uint16_t*)(v.packet.data()))) == AUDIO_TRACK_MARKER;
```

Comment:
Actually it looks like we still have indentation issues with this lambda here, there is an extra tab

Reply:
it's default vim indent behavior with correct shiftwidth and tabstop, maybe we need to create .clang-format file to make formatting easier across different editor

Reply:
Use a real IDE like Visual Studio ![catsuit](https://cdn.discordapp.com/emojis/1156932934306119711.png?size=16&quality=lossless)

Reply:
i mean u can just export the built-in formatting config of VS into an actual .clang-format file r8? surely such "full featured real IDE" like VS can do a simple export ![image](https://camo.githubusercontent.com/827c27a35b02e389a92e1b5967bf0e1d1595a23437900cc04c47c614837bac66/68747470733a2f2f63646e2e646973636f72646170702e636f6d2f656d6f6a69732f313135363933323933343330363131393731312e706e673f73697a653d3136267175616c6974793d6c6f73736c657373) ![image](https://camo.githubusercontent.com/827c27a35b02e389a92e1b5967bf0e1d1595a23437900cc04c47c614837bac66/68747470733a2f2f63646e2e646973636f72646170702e636f6d2f656d6f6a69732f313135363933323933343330363131393731312e706e673f73697a653d3136267175616c6974793d6c6f73736c657373) ![image](https://camo.githubusercontent.com/827c27a35b02e389a92e1b5967bf0e1d1595a23437900cc04c47c614837bac66/68747470733a2f2f63646e2e646973636f72646170702e636f6d2f656d6f6a69732f313135363933323933343330363131393731312e706e673f73697a653d3136267175616c6974793d6c6f73736c657373)